src/sentry/tasks/commit_context.py

```diff
@@ -44,19 +45,35 @@ def queue_comment_task_if_needed(
         extra={"organization_id": commit.organization_id, "merge_commit_sha": commit.key},
     )
 
+    # client will raise ApiError if the request is not successful
     response = installation.get_client().get_pullrequest_from_commit(repo=repo.name, sha=commit.key)
 
-    if not (response.status_code == 200 and isinstance(response, list) and len(response) == 1):
+    if not (isinstance(response, list) and len(response) == 1):
```

Comment:
Can we change this to be `if not isinstance(response, list) or len(response) !=1`?  Applying the negation to the operation can be misleading (for example I thought the check on line 53 was redundant at first).